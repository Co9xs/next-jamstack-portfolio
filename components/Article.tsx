import styled from 'styled-components'
export const Article = ({ article }) => { 
  const convertToDate = (dt: Date) => {
    const year = dt.getFullYear();
    const month = ("00" + (dt.getMonth()+1)).slice(-2);
    const date = ("00" + dt.getDate()).slice(-2);
    const result = `${year}/${month}/${date}`
    return result;
  }
  const publishedAt = convertToDate(new Date(article.publishedAt));
  return (
    <ArticleBase>
      <ArticleHeader>
        <ArticleTitle>{ article.title }</ArticleTitle>
      </ArticleHeader>
      <ArticleFooter>
        <ArticleDate>{ publishedAt }</ArticleDate>
        {article.category && <ArticleCategory>#{ article.category.name }</ArticleCategory>}
      </ArticleFooter>
    </ArticleBase>
  )
}

const ArticleBase = styled.article`
  padding: 0.75rem;
  border: 2px solid #EFEFEF;
  border-radius: 5px;
  padding: 1.5rem;
  margin-bottom: 1.5rem;
  background: #FFF;
  cursor: pointer;

  //DarkMode
  color: ${({ theme }) => theme.white};
  background: ${({ theme }) => theme.black};
  border-color: ${({ theme }) => theme.smoke};
`

const ArticleHeader = styled.div`
  display: flex;
`
const ArticleTitle = styled.h3`
  margin: 0;
  padding: 0;
`
const ArticleFooter = styled.div`
`
const ArticleCategory = styled.span`
  color: #60A5FA;
  margin-right: .5rem;
`
const ArticleDate = styled.span`
  color: #6B7280;
  margin-right: 1rem;

  color: ${({ theme }) => theme.smoke};
`