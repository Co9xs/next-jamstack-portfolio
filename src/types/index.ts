export type Category = {
  id: string,
  name: string,
  articleCount?: number
}

export type CategoryData = {
  contents: Category[],
  totalCount: number
}

export type Article = {
  id: string,
  title: string,
  body: string,
  publishedAt: string,
  category: Category | null
}

export type ArticleData = {
  contents: Article[],
  totalCount: number
}

export type Work = {
  title: string,
  date: string,
  url: string,
  githubUrl: string,
  languages: string[],
  description: string
}

export type Skill = {
  name: string
}