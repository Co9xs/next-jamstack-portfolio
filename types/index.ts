export type Category = {
  name: string
}

export type Article = {
  id: string,
  title: string,
  body: string,
  description: string,
  publishedAt: string,
  category: Category | null
}