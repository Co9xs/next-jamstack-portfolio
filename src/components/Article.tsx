import styled from 'styled-components'
import { Article } from '@/types';
import { convertDateToString } from '@/utils';
import React from 'react';
import { ClockIcon, PersonIcon } from './icons';

type Props = {
  article: Article
}

export const ArticleCard: React.VFC<Props> = (props) => {
  const { article } = props
  const publishedAt = convertDateToString(new Date(article.publishedAt));
  console.log(article.body.length)
  return (
    <ArticleBase>
      <ArticleImage>
      </ArticleImage>
      <ArticleData>
        <ArticleTitle>{ article.title }</ArticleTitle>
        <ArticleCategory>{article.category.name}</ArticleCategory>
        <ArticleTags>
          {article.tags.map(tag => (
            <ArticleTag key={tag.id}>#{tag.name}</ArticleTag>
          ))}
        </ArticleTags>
        <ArticleMetaData>
          <ArticleDate>
            <ClockIcon />
            <ArticleDateText>
              {publishedAt}
            </ArticleDateText>
          </ArticleDate>
          <ArticleReadTime>30 min read</ArticleReadTime>
          <ArticleAuthor>
            <PersonIcon/>
            <ArticleAuthorText>{ article.author?.displayName}</ArticleAuthorText>
          </ArticleAuthor>
        </ArticleMetaData>
      </ArticleData>
    </ArticleBase>
  )
}

const ArticleBase = styled.article`
  padding: 16px 0;
  cursor: pointer;
  display: flex;
  justify-content: space-between;

  //DarkMode
  color: ${({ theme }) => theme.white};
  background: ${({ theme }) => theme.black};
  border-color: ${({ theme }) => theme.smoke};
`

const ArticleImage = styled.div`
  width: 350px;
  min-width: 250px;
  height: 175px;
  background-color: #EEE;
  border-radius: 3px;
  background-size: cover;
  background-position: center;
`

const ArticleData = styled.div`
  min-width: 420px;
  max-width: 420px;
`
const ArticleTitle = styled.h3`
  margin: 0;
  padding: 0;
  font-size: 20px;
  margin-bottom: 10px;
`

const ArticleCategory = styled.span`
  color: #60A5FA;
  padding: 4px 8px;
  border: 2px solid #60A5FA;
  border-radius: 3px;
  display: inline-block;
  margin-bottom: 10px;
  font-size: 14px;
`

const ArticleTags = styled.div`
  color: #616269;
  margin-bottom: 10px;
`

const ArticleTag = styled.span`
  margin-right: 8px;
`

const ArticleMetaData = styled.div`
  color: #616269;
  display: flex;
  align-items: center;
`

const ArticleDate = styled.span`
  margin-right: 16px;
  display: flex;
  align-items: center;
  color: ${({ theme }) => theme.smoke};
`

const ArticleDateText = styled.span`
  margin-left: 8px;
`

const ArticleReadTime = styled.div`
  margin-right: 16px;
`

const ArticleAuthor = styled.div`
  display: flex;
  align-items: center;
`

const ArticleAuthorText = styled.span`
  margin-left: 8px;
`