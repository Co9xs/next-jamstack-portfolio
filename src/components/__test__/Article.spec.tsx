import { Article } from '@/types';
import React from 'react'
import renderer from 'react-test-renderer'
import { ArticleCard } from '../Article';

const mockArticle: Article = {
  id: 'job-hunting-entry',
  publishedAt: '2021-04-21T08:07:07.339Z',
  title: '【テスト記事】テスト記事です',
  body: '<h2 id="h8d027c8ed3">テスト記事</h2>',
  category: {
    id: 'tech',
    name: 'Tech'
  }
}

test('ArticleCard', () => {
  const component = renderer.create(<ArticleCard article={mockArticle}/>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot();
})