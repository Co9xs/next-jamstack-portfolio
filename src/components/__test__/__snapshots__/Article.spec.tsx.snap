// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard 1`] = `
<article
  className="sc-bdfBwQ cIcVTu"
>
  <div
    className="sc-gsTCUz ywWHk"
  >
    <h3
      className="sc-dlfnbm haUCCe"
    >
      【テスト記事】テスト記事です
    </h3>
  </div>
  <div
    className="sc-hKgILt chrFSH"
  >
    <span
      className="sc-jSgupP jvTTyp"
    >
      2021/04/21
    </span>
    <span
      className="sc-eCssSg iCSNRt"
    >
      #
      Tech
    </span>
  </div>
</article>
`;
